### PR TITLE
Examine Tcomms machinery to see damage

### PR DESCRIPTION
**Pull request by EvenInDeathIStillServe**
TComms machinery can now be examined to know if they are damaged or just unpowered. Different message depending on severity of damage.

### CODE REVIEW

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 1]
```diff
@@ -14,6 +14,23 @@
...
+		if(65 to 85)
+			user << "It's slightly damaged."
+		if(45 to 65)
+			user << SPAN_WARNING("It's badly damaged.")
+		if(25 to 45)
+			user << SPAN_WARNING("It's heavily damaged.")
+		else
+			user << SPAN_WARNING("It's falling apart.")
+	if(stat & BROKEN)
+		user << SPAN_DANGER("It's broken.")
```
Comment (LethalGhost) [inline code/game/machinery/telecomms/machine_interactions.dm]: Isn't it better to check `stat` first and if it's not `BROKEN` check `integrity`?
That seems right now you can get message like _"This is telecomms machinry. It's in perfect condition It's broken."_

**Comment by LagannPM** [comment]
Isn't it better to move this code to code/game/machinery instead ?

**Comment by EvenInDeathIStillServe** [comment]
No, because integrity is a variable specific to Telecomms machinery. Autolathes, gibbers, etc do not get damaged.

File: code/game/machinery/telecomms/machine_interactions.dm (hunk 1 above)
Comment (EvenInDeathIStillServe) [reply code/game/machinery/telecomms/machine_interactions.dm]: Tcomms machinery has two separate damage conditions: integrity, which is caused by overheating, and the BROKEN stat, which only happens when you remove cables from it while disassembling it.

I have changed the broken message to better reflect its status.

**Comment by LagannPM** [comment]
We actually need a destroyable autolathes, gibbers, etc. , if this not a problem for you can you move integrity variable, examine code, harm intent interactions and repair code to /obj/machinery instead ?

**Comment by EvenInDeathIStillServe** [comment]
That is acceptable. I'll see what I can do.

**Comment by NanakoAC** [comment]
@ITCups @EvenInDeathIStillServe Please give me a status report on this PR

**Comment by EvenInDeathIStillServe** [comment]
Haven't done anything other than my initial commit. I can try to get it done in about a week.

**Comment by NanakoAC** [comment]
requested closed by author, hope he'll have time to work on it again soon

**Pull request closed without merging**